packages/tokens/src/typography.ts

```diff
@@ -1,101 +1,246 @@
-const marchePro = 'Marche';
-
-const commonMarcheProStyles = {
-  fontFamily: marchePro,
-  textTransform: 'uppercase',
+const fontMap = {
+  web: {
+    firaCode: 'Fira Code',
+    diatype: 'Diatype',
+    marchePro: 'Marche',
+  },
+  mobile: {
+    firaCode: 'Fira Code',
+    diatype: 'DiatypeRegular',
+    marchePro: 'MarcheSuperPro',
+  },
 };
 
-const diatype = 'Diatype';
+// we might want to split up mobile into ios and android in the future.
+type Platform = 'mobile' | 'web';
 
-const commonDiatypeStyles = {
-  fontFamily: diatype,
-};
+function getTextVariants({ platform }: { platform: Platform }) {
+  const commonMarcheProStyles = {
+    fontFamily: fontMap[platform].marchePro,
+    textTransform: 'uppercase',
+  };
 
-const firaCode = 'Fira Code';
+  const commonDiatypeStyles = {
+    fontFamily: fontMap[platform].diatype,
+  };
 
-export const textStyles = {
-  'display.01': {
-    description: 'display.01',
-    value: { ...commonMarcheProStyles, fontSize: '9.375rem', lineHeight: '7.5rem' },
-  },
-  'display.02': {
-    description: 'display.02',
-    value: { ...commonMarcheProStyles, fontSize: '4rem', lineHeight: '3.5rem' },
-  },
+  const commonFiracodeStyles = {
+    fontFamily: fontMap[platform].firaCode,
+  };
 
-  'heading.01': {
-    description: 'heading.01',
-    value: { ...commonDiatypeStyles, fontSize: '3.3125rem', lineHeight: '3.75rem' },
-  },
-  'heading.02': {
-    description: 'heading.02',
-    value: { ...commonMarcheProStyles, fontSize: '2.75rem', lineHeight: '2.75rem' },
-  },
-  'heading.03': {
-    description: 'heading.03',
-    value: { ...commonMarcheProStyles, fontSize: '2rem', lineHeight: '2.1875rem' },
-  },
-  'heading.04': {
-    description: 'heading.04',
-    value: {
-      ...commonDiatypeStyles,
-      fontSize: '1.625rem',
-      lineHeight: '2.25rem',
-      fontWeight: '500',
+  function transformSize<T extends number>(size: T) {
+    switch (platform) {
+      case 'web':
+        return `${(size / 16).toFixed(4)}rem` as const;
+      case 'mobile':
+        return size;
+    }
+  }
+
+  function transformWeight<T extends number>(weight: T) {
+    switch (platform) {
+      case 'web':
+        return weight;
+      case 'mobile':
+        return weight.toString();
+    }
+  }
+
+  const display01 = {
```

Comment:
Should we take on this issue with these updates? https://github.com/leather-wallet/mono/issues/72